Sources/LiveViewNative/Views/Controls and Indicators/Links/ShareLink.swift

```diff
@@ -0,0 +1,328 @@
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item), image: images.value(for: item))
+                    )
+                case let .iconOnly(titles, icons):
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item), icon: icons.value(for: item))
+                    )
+                }
+            } else {
+                switch previews(for: [item]) {
+                case nil:
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message
+                    ) { label }
+                case let .titled(titles):
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item))
+                    ) { label }
+                case let .complete(titles, images, icons):
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item), image: images.value(for: item), icon: icons.value(for: item))
+                    ) { label }
+                case let .imageOnly(titles, images):
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item), image: images.value(for: item))
+                    ) { label }
+                case let .iconOnly(titles, icons):
+                    SwiftUI.ShareLink(
+                        item: item,
+                        subject: subject,
+                        message: message,
+                        preview: .init(titles.value(for: item), icon: icons.value(for: item))
+                    ) { label }
+                }
+            }
+        }
+    }
+    
+    private var label: some View {
+        context.buildChildren(of: element, withTagName: "label", namespace: "share-link", includeDefaultSlot: true)
+    }
+    
+    /// The set of values used to create the `SharePreview` for each item.
+    private enum PreviewBranch<Item: Hashable> {
+        struct Properties<Property> {
+            /// A map between share items and a certain property of the `SharePreview`.
+            let values: [Item:Property]
+            /// The value in the default `SharePreview`.
+            let defaultValue: Property?
+            
+            @inlinable
```

Comment:
I don't think marking these inlinable does anything, since they're module-internal anyway. If you want them to be inlined, `@inline(__always)` is what you're after.

Reply:
Oops, thanks. Best to leave it to the compiler.